indico/modules/events/papers/forms.py

```diff
@@ -145,3 +188,42 @@ def __init__(self, *args, **kwargs):
         if template is None:
             inject_validators(self, 'template', [DataRequired()])
         super(PaperTemplateForm, self).__init__(*args, **kwargs)
+
+
+class PaperCommentForm(IndicoForm):
+    text = TextAreaField(_("Comment"), [DataRequired()], render_kw={'placeholder': _("Leave a comment...")})
+    visibility = IndicoEnumSelectField(_("Visibility"), [DataRequired()], enum=PaperCommentVisibility,
+                                       skip={PaperCommentVisibility.users})
+
+    def __init__(self, *args, **kwargs):
+        comment = kwargs.get('obj')
+        user = comment.user if comment else kwargs.pop('user')
+        paper = kwargs.pop('paper')
+        super(IndicoForm, self).__init__(*args, **kwargs)
+        if not paper.can_judge(user):
+            self.visibility.skip.add(PaperCommentVisibility.judges)
+        if not paper.can_review(user):
+            del self.visibility
+
+
+class PaperReviewForm(IndicoForm):
+    """Form for reviewing a paper"""
+
+    _order = ('proposed_action', 'comment')
```

Comment:
Why do we need to specify an order here? Usually we do this when extending the from a base class, but that's not the case.